libs/wingsdk/src/shared-aws/function.inflight.ts

```diff
@@ -11,12 +13,49 @@ export class FunctionClient implements IFunctionClient {
   /**
    * Invoke the function, passing the given payload as an argument.
    */
-  public async invoke(payload: string): Promise<string> {
+  public async invoke(
+    payload: string,
+    logsCollector?: Trace[]
+  ): Promise<string> {
     const command = new InvokeCommand({
       FunctionName: this.functionArn,
       Payload: fromUtf8(JSON.stringify(payload)),
+      LogType: LogType.Tail,
     });
     const response = await this.lambdaClient.send(command);
+    const arnParts = this.functionArn.split(":");
+    const functionName = arnParts[arnParts.indexOf("function") + 1];
+
+    if (logsCollector) {
+      const logs: string[] =
+        atob(response.LogResult ?? "").match(
+          // date, then any char then a line of winglog
+          /(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z)(.*)winglogstart:[\s\S]+?:winglogend/gm
+        ) ?? [];
+
+      logsCollector.push(
+        ...logs.map((item: string) => ({
+          data: {
+            message: (item.match(/(?<=winglogstart:)[\s\S]*?(?=:winglogend)/) ??
+              [])[0],
+          },
+          sourceType: FUNCTION_TYPE,
+          sourcePath: functionName, //TODO: is that the right thing to do?
+          type: TraceType.LOG,
+          timestamp:
+            (item.match(/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/) ??
+              [])[0] ?? "non-existent",
+        }))
+      );
+    } else {
+      console.log(`Invoking ${functionName}:`);
+      console.log(
+        atob(response.LogResult ?? "")
+          .match(/winglogstart:[\s\S]+?:winglogend/gm)
+          ?.join("\n") ?? ""
+      );
+    }
```

Comment:
Do we need this else branch?

Reply:
Yes, when we have a function that is called in another function or a test, it doesn't get caught, unless it prints to the test function log